### ISSUE TITLE
Feature Request: Support for distributed training with mirrored strategy (data parallel)

### ISSUE THREAD

**Issue by piEsposito** [tensorflow/probability#1469]
We are currently unable to perform distributed training using TFP Reparameterized Layers due to the weight sampling being done differently across all workers (that decreases a lot the learning of the model across the epochs). 

It would be nice if we had a way to ensure a specific "reparameterized weight sampling" random seed is the same across every worker (and GPU) on data-parallel scenarios.

**Comment by brianwa84** [comment]
To get the same seed across devices, you can use a Tensor seed,
tfd.Normal(0, 1).sample(seed=tf.constant([0, 1])).
You can use tfp.experimental.distribute.Sharded(underlying_dist) for an
event that's physically partitioned across devices.
tfp.experimental.distribute.JointDistribution* supports the case of global
(cross-device replicated) and per-device (Sharded) latents.

None of the above ties in super-elegantly with the reparameterized
tfp.layers, though tfp.experimental.joint_distribution_layers might be
interesting here, as might tfp.experimental.nn.


Brian Patton | Software Engineer | ***@***.***
